frontend/src/pages/EditProfile.tsx

```diff
@@ -0,0 +1,102 @@
+import { useState } from 'react';
+import './EditProfile.css';
+import useAuth from './UserAuth';
+
+function EditProfile() {
+  const { username: username1 } = useAuth();
+
+  const [gender, setGender] = useState('');
+  const [occupation, setOccupation] = useState('');
+  const [favoriteBook, setFavoriteBook] = useState('');
+  const [description, setDescription] = useState('');
+  const [username, setusername] = useState('');
+  const [isSaved, setIsSaved] = useState(false);
+
+  const handleSave = async () => {
+    try {
+      if (username.trim() === '') {
+        return alert('Username cannot be empty');
+      }
+      if (username.trim() !== username1) {
+        return alert('Username is not correct');
+      }
+      const response = await fetch(
+        'http://localhost:3001/api/saveProfileData',
+        {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json'
+          },
+          body: JSON.stringify({
+            gender,
+            occupation,
+            favoriteBook,
+            description,
+            username
+          })
+        }
+      );
```

Comment:
This works fine locally, but the hard-coded url will cause issues when we deploy to production.

Could we make this into a function in `frontend/src/services/index.ts` instead?
```ts
// Example
export async function saveProfileData(profileData: ProfileData) {
  return await fetch(`${BASE_URL}/api/saveProfileData`, ...)  // etc
}
```